### ISSUE TITLE
Calling some functions decorated with `wrap_ad` results in a crash

### ISSUE THREAD

**Issue by mworchel** [mitsuba-renderer/drjit#110]
Hi,

I discovered a pair of settings where wrapping a function with `wrap_ad(source='torch', target='drjit')` and calling it results in a crash and you might be interested in looking into it. Some of it might be expected behavior (?) although I don't see why it should be. Here are minimal reproducers.

1) The identity function
```python
import drjit as dr
import torch

device = torch.device('cuda:0')

@dr.wrap_ad(source='torch', target='drjit')
def do_something(u):
    return u

u = torch.tensor([0], dtype=torch.float32, device=device)
v = do_something(u) # crashes here
```

2) Taking the gradient of a simple sum involving the input and any number
```python
import drjit as dr
import torch

device = torch.device('cuda:0')

@dr.wrap_ad(source='torch', target='drjit')
def do_something(u):
    return u + 1

u = torch.tensor([0], dtype=torch.float32, device=device)
u.requires_grad = True

v = do_something(u)

du = torch.autograd.grad(v.mean(), u)[0] # crashes here
```

Transforming the input with an arbitrary `drjit` function fixes the crashes, for example by defining `do_something` as 

```python
def do_something(u):
    return dr.abs(u) + 1
```

Here some information about my system:
- OS: Windows 10
- GPU: RTX 3080 Ti
- Python 3.8.10
  - `drjit 0.3.2`
  - `pytorch 1.12.1`
  - `cudatoolkit 11.6.0`

**Comment by njroussel** [comment]
Hi @mworchel 

Thanks for the reproducers. There are already a few different open issues concerning `wrap_ad` at the moment, I've also added this to our to-do list. We'll eventually get around to fixing them all :)

**Comment by memamsaleh** [comment]
Hi @njroussel 

Any updates on the `wrap_ad` issues? It crashes even for the examples in the official documentation.

**Comment by Speierers** [comment]
The tutorial runs fine on my end (and on the CI). Could you please be more specific regarding the crash? And maybe share your system specs?

**Comment by memamsaleh** [comment]
Hi @Speierers 

The mitsuba tutorial does work fine for me as well, however the two simple code snippets related to `wrap_ad` in the [drjit api reference](https://drjit.readthedocs.io/en/latest/reference.html#drjit.wrap_ad) result in a crash. The crash happens when calling the decorated function in both cases.

System Specs:
  OS: Windows-10
  CPU: AMD64 Family 25 Model 33 Stepping 2, AuthenticAMD
  GPU: NVIDIA GeForce RTX 4090
  Python: 3.11.4 | packaged by Anaconda, Inc. | [MSC v.1916 64 bit (AMD64)]
  NVidia driver: 537.13
  CUDA: 12.2.91
  LLVM: -1.-1.-1

  Dr.Jit: 0.4.3
  Mitsuba: 3.4.0
     Is custom build? False
     Compiled with: MSVC 19.34.31937.0
     Variants:
        scalar_rgb
        cuda_ad_rgb
        cuda_rgb
        cuda_mono_polarized
        cuda_ad_mono_polarized

I have also tested with the pre-built binaries (pip installer) in a separate environment and faced the same issues.

**Comment by njroussel** [comment]
Hi @memamsaleh 

I think this is another case of PyTorch's stupid symbol leaking. If you're building `drjit` yourself, the safest approach is to have `torch` as your very first import.

I was able to verify this on both the two original reproduces of this issue and the snipped in the API reference.

 > I have also tested with the pre-built binaries (pip installer) in a separate environment and faced the same issue

However, this should have worked I believe. 

The good news is that someone finally fixed this over in PyTorch: https://github.com/pytorch/pytorch/commit/ddff4efa26d527c99cd9892278a32529ddc77e66 :tada: 
As far as I can tell, it will be part of the torch 2.1 release which should be available in October.
